.github/workflows/clang-tidy.yml

```diff
@@ -32,36 +51,87 @@ concurrency:
+        id: get-versions
+        run: |
+          content=`cat ${{ github.workspace }}/oneDNN/.github/automation/aarch64/ci.json`
+          content="${content//[$'\t\r\n$ ']}"
+          echo "output=$content" >> $GITHUB_OUTPUT
+
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Clone ACL
+        run: ${{ github.workspace }}/oneDNN/.github/automation/aarch64/build_acl.sh
+        env:
+          ACL_ACTION: clone
+          ACL_ROOT_DIR: ${{ github.workspace }}/ComputeLibrary
+          ACL_VERSION: ${{ fromJson(steps.get-versions.outputs.output).dependencies.acl }}
 
-      - name: Configure oneDNN
+      # We can avoid having to actually build ComputeLibrary by faking the files
+      # that cmake/FindACL.cmake and cmake/ACL.cmake expect to see.
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Fake ACL Build
+        working-directory: ${{ github.workspace }}/ComputeLibrary/
+        env:
+          ACL_VERSION: ${{ fromJson(steps.get-versions.outputs.output).dependencies.acl }}
+        run: |
+          mkdir ./build
+          echo "arm_compute_version=${ACL_VERSION}" > ./build/arm_compute_version.embed
+          touch ./build/libarm_compute.so
+          touch ./build/libarm_compute_graph.so
+
+      - if: ${{ matrix.config.arch == 'x64' }}
+        name: Configure oneDNN - x64
+        working-directory: ${{github.workspace}}/oneDNN
         run: .github/automation/x64/build_linters.sh
         env:
           ONEDNN_ACTION: configure
 
+      - if: ${{ matrix.config.arch == 'AArch64' }}
+        name: Configure oneDNN - AArch64
+        working-directory: ${{github.workspace}}/oneDNN
+        run: .github/automation/aarch64/build.sh
+        env:
+          BUILD_TOOLSET: clang
+          ACL_ROOT_DIR: ${{github.workspace}}/ComputeLibrary
+          ONEDNN_ACTION: configure
+
       - name: Check source files
+        working-directory: ${{github.workspace}}/oneDNN
+        shell: bash
         run: |
+          set +e
           echo -e "Checking Clang-Tidy $(clang-tidy --version)\n"
           touch clang-tidy.log
-          for file in $(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.cpp$');
-          do
-            if grep -q "$file" "build/compile_commands.json"; then
-              echo -e "\nAnalyzing $file"
-              clang-tidy -p build $file 2>&1 | tee -a clang-tidy.log
-            else
-              echo "Skipped $file as it's not built in x64 OpenMP/OpenCL configuration."
-            fi
-          done
+          analysis_list="$(git diff --name-only ${{ github.event.pull_request.base.sha }} | grep -E '\.[ch](pp|xx)?$')"
+          if [[ -z "$analysis_list" ]]; then
+            echo "No source files to check."
+            exit 0
+          else
+            printf "File list:\n$analysis_list\n"
+            run-clang-tidy -j $(nproc) -format -config-file=./.clang-tidy -p ./build $analysis_list 2>&1 | tee -a clang-tidy.log
```

Comment:
`-config-file=./.clang-tidy` is not needed, as clang-format will pick up closest .clang-tidy from the hierarchy. oneDNN has multiple `.clang-tidy` files in the source tree.

Reply:
Hmm I tried making this change locally but it throws errors from trying to read clang-tidy configs outside oneDNN (i.e, the ComputeLibrary repository). I have uploaded the change regardless so we can see if it causes issues in the CI.

Reply:
Huh seems to work fine. I guess something is strange with my local environment. Thanks for the suggestion!